packages/google_maps_flutter/google_maps_flutter_web/lib/src/circles.dart

```diff
@@ -0,0 +1,79 @@
+part of google_maps_flutter_web;
+
+/// This class manages all the circles associated to any given Google Map Controller.
+class CirclesController extends AbstractController {
+  // A cache of circleIDs to their controllers
+  final Map<CircleId, CircleController> _circleIdToController;
+
+  // The stream over which circles broadcast their events
+  StreamController<MapEvent> _streamController;
+
+  /// Initialize the cache. The StreamController is shared with the Google Map Controller.
+  CirclesController({
+    @required StreamController<MapEvent> stream,
+  })  : _streamController = stream,
+        _circleIdToController = Map<CircleId, CircleController>();
+
+  /// Returns the cache of circles. Test only.
+  @visibleForTesting
+  Map<CircleId, CircleController> get circles => _circleIdToController;
+
+  /// Adds a set of [Circle] objects to the cache.
+  ///
+  /// (Wraps each Circle into its corresponding [CircleController])
+  void addCircles(Set<Circle> circlesToAdd) {
+    circlesToAdd?.forEach((circle) {
+      _addCircle(circle);
+    });
+  }
+
+  void _addCircle(Circle circle) {
+    if (circle == null) return;
+    final populationOptions = _circleOptionsFromCircle(circle);
+    gmaps.Circle gmCircle = gmaps.Circle(populationOptions);
+    gmCircle.map = googleMap;
+    CircleController controller = CircleController(
+        circle: gmCircle,
+        consumeTapEvents: circle.consumeTapEvents,
+        onTap: () {
+          _onCircleTap(circle.circleId);
+        });
+    _circleIdToController[circle.circleId] = controller;
+  }
+
+  /// Updates a set of [Circle] objects with new options.
+  void changeCircles(Set<Circle> circlesToChange) {
+    circlesToChange?.forEach((circleToChange) {
+      _changeCircle(circleToChange);
+    });
+  }
+
+  void _changeCircle(Circle circle) {
+    if (circle != null) {
+      final circleController = _circleIdToController[circle.circleId];
+      circleController?.update(_circleOptionsFromCircle(circle));
+    }
+  }
+
+  /// Removes a set of [CircleId]s from the cache.
+  void removeCircles(Set<CircleId> circleIdsToRemove) {
+    circleIdsToRemove?.forEach((circleId) {
+      if (circleId != null) {
```

Comment:
`Set<CircleId> circleIds = { null };` is valid dart, and I don't know how this will be called.

`assert` `!= null` is more idiomatic. The problem I have with it is that asserts don't prevent run-time failures in prod, so I preferred to leave these sanity checks in. WDYT?

Reply:
I just made the internal code null-safe, so this probably doesn't matter :)